syncd/syncd_notifications.cpp

```diff
@@ -461,8 +461,70 @@ void processNotification(
 // that some notiffication arrived
 
 std::condition_variable cv;
-std::mutex queue_mutex;
-std::queue<swss::KeyOpFieldsValuesTuple> ntf_queue;
+
+class ntf_queue_t
+{
+public:
+    bool enqueue(swss::KeyOpFieldsValuesTuple msg);
+    bool tryDequeue(swss::KeyOpFieldsValuesTuple& msg);
+    size_t queueStats()
+    {
+        std::lock_guard<std::mutex> lock(queue_mutex);
+
+        return ntf_queue.size();
+    }
+
+private:
+    std::mutex queue_mutex;
+    std::queue<swss::KeyOpFieldsValuesTuple> ntf_queue;
+    const size_t limit = 300000;
+};
+
+static std::unique_ptr<ntf_queue_t> ntf_queue_hdlr;
+
+bool ntf_queue_t::tryDequeue(
+        _Out_ swss::KeyOpFieldsValuesTuple &item)
+{
+    std::lock_guard<std::mutex> lock(queue_mutex);
+
+    SWSS_LOG_ENTER();
+
+    if (ntf_queue.empty())
+    {
+        return false;
+    }
+
+    item = ntf_queue.front();
+
+    ntf_queue.pop();
+
+    return true;
+}
+
+bool ntf_queue_t::enqueue(
+        _In_ swss::KeyOpFieldsValuesTuple item)
+{
+    SWSS_LOG_ENTER();
+
+    std::string notification = kfvKey(item);
+
+    /*
+     * If the queue exceeds the limit, then drop all further FDB events
+     * This is a temporary solution to handle high memory usage by syncd and the
+     * ntf-Q keeps growing. The permanent solution would be to make this stateful
+     * so that only the *latest* event is published.
+     */
+    if (queueStats() < limit || notification != "fdb_event")
+    {
+        std::lock_guard<std::mutex> lock(queue_mutex);
+
+        ntf_queue.push(item);
+        return true;
+    }
+
+    SWSS_LOG_INFO("Too many messages in queue (%ld), dropping FDB events!", queueStats());
```

Comment:
this should be info, can we add a counter (static) and print out message once every 100 packets.

Reply:
sorry, I mean the level should be notice